tests/phpunit/CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor/LineItemTest.php

```diff
@@ -23,8 +23,17 @@ class CRM_MembershipExtras_Hook_Pre_MembershipPaymentPlanProcessor_LineItemTest
   use CRM_MembershipExtras_Test_Helper_FinancialAccountTrait;
   use CRM_MembershipExtras_Test_Helper_FixedPeriodMembershipTypeSettingsTrait;
 
-  private $membrship;
   private $priceSet;
+  /**
+   * @var mixed
+   */
+  private $membership;
+
+  public function setUp() {
+    //We mocked membership ID as the membership ID is always when
+    //Contribution pre_process hook is called.
+    MembershipPaymentPlanProcessor::$membership_id = $this->membership['id'];;
```

Comment:
Should we reset it after the test using the `teardown` method ?

Reply:
@ahed-compucorp  perhaps, we don't need it as we always reset it when we setup, it is not like we need to tear down environment variables.